cni/pkg/ambient/server.go

```diff
@@ -170,18 +193,47 @@ func (s *Server) ReconcileZtunnel() error {
 		return nil
 	}
 	log.Infof("active ztunnel updated to %v", activePod.Name)
-	// TODO: we should not cleanup and recreate; this has downtime. We should mutate the existing rules in place
-	s.cleanup()
-	veth, err := getDeviceWithDestinationOf(activePod.Status.PodIP)
-	if err != nil {
-		return fmt.Errorf("failed to get device: %v", err)
-	}
 
 	captureDNS := getEnvFromPod(activePod, "ISTIO_META_DNS_CAPTURE") == "true"
-	err = s.CreateRulesOnNode(veth, activePod.Status.PodIP, captureDNS)
-	if err != nil {
-		return fmt.Errorf("failed to configure node for ztunnel: %v", err)
+
+	switch s.redirectMode {
+	case IptablesMode:
+		// TODO: we should not cleanup and recreate; this has downtime. We should mutate the existing rules in place
+		s.cleanup()
+		veth, err := getDeviceWithDestinationOf(activePod.Status.PodIP)
+		if err != nil {
+			return fmt.Errorf("failed to get device: %v", err)
+		}
+		err = s.CreateRulesOnNode(veth, activePod.Status.PodIP, captureDNS)
+		if err != nil {
+			return fmt.Errorf("failed to configure node for ztunnel: %v", err)
+		}
+	case EbpfMode:
+		h, err := GetHostIPByRoute(s.kubeClient.Kube())
```

Comment:
at this point we already have a Pod for ztunnel, no need to query anything in GetHostIPByRoute